File: src/main/java/dev/ambryn/task/models/Running.java
package dev.ambryn.task.models;

import dev.ambryn.task.interfaces.State;

import java.time.LocalDateTime;

public class Running extends State {

    public Running(Task context) {
        this.context = context;
        this.status = Status.RUNNING;
    }

    @Override
    public void start() {
        return;
    }

    @Override
    public void error() {
        this.context.setState(new Error(context));
    }

    @Override
    public void finish() {
        this.context.setState(new Finished(context));
        this.context.setFinishedAt(LocalDateTime.now());
    }

    @Override
    public void cancel() {
        this.context.setState(new Cancelled(context));
    }
}

File: src/main/java/dev/ambryn/task/models/Running.kt
package dev.ambryn.task.models

import dev.ambryn.task.interfaces.State
import java.time.LocalDateTime

class Running(context: Task) : State(context) {
    override var status: Status = Status.RUNNING

    override fun error() {
        context.state = Errored(context)
    }

    override fun finish() {
        context.state = Finished(context)
        context.finishedAt = LocalDateTime.now()
    }

    override fun cancel() {
        context.state = Cancelled(context)
    }
}
